### PR TITLE
[jenkins] Build android as RelWithDebInfo by default

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

Build kodi for android as release by default
the goal is to have nightlies builds as release

## Motivation and context

Dscussed some time ago on slack about the problems with subtitles and other GUI slowdown with android nightly builds because are builded as debug

So for user testing is better have nightly builds as release type as already happens with Windows nightly builds

code change suggested by @fuzzard

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [x] **None of the above** (please explain below)
Build type change

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by a1rwulf** [comment]
Shouldn't it be `RelWithDebInfo`?

**Comment by CastagnaIT** [comment]
changed to RelWithDebInfo

File: tools/buildsteps/defaultenv [hunk 1]
```diff
@@ -59,7 +59,7 @@ case $XBMC_PLATFORM_DIR in
     DEFAULT_NDK_VERSION="21e" # NDK package version (newer API can be inside)
     DEFAULT_NDK_API="21" # Lollipop API level (21) defined in package ./sysroot/usr/include/android/api-level.h
     DEFAULT_XBMC_DEPENDS_ROOT=$WORKSPACE/tools/depends/xbmc-depends
-    DEFAULT_CONFIGURATION="Debug"
+    DEFAULT_CONFIGURATION="RelWithDebInfo"
```
Comment (lrusak) [inline tools/buildsteps/defaultenv:62]: Should this be the default for all platforms?

**Comment by fuzzard** [comment]
Cherry pick https://github.com/fuzzard/xbmc/commit/d73f810f0a78b264826e7f588f8a52bdd3a888d0 and add as the first commit

**Comment by fuzzard** [comment]
Bloody case sensitive FS. Try this one. both worked fine on a macos host :/

https://github.com/fuzzard/xbmc/commit/2604a8df76579c521f761ebade308149db2de56b

**Comment by wsnipex** [comment]
shouldn't we simply adjust the jenkins job config instead?

**Comment by fuzzard** [comment]
Jenkins jobs just use default unless overridden

File: tools/buildsteps/defaultenv (hunk 1 above)
Comment (CastagnaIT) [reply tools/buildsteps/defaultenv:62]: for me no problem set it for all them
but i am not the best person to ask
@fuzzard ?

Comment (fuzzard) [reply tools/buildsteps/defaultenv:62]: I dont really have an opinion either way.

Id probably just go as is for now until we find a need on other platforms. Most other platforms are more performant than android systems, so we may not really see any difference either way.

**Comment by fuzzard** [comment]
Lets just go with this. If any other platforms need to change, they can always do it as needed.

**Pull request merged**